libs/terraform/plan.go

```diff
@@ -1,13 +1,40 @@
 package terraform
 
+import "strings"
+
 type Plan struct {
 	// Path to the plan
 	Path string
 
-	// Holds whether the user can consented to destruction. Either by interactive
-	// confirmation or by passing a command line flag
-	ConfirmApply bool
-
 	// If true, the plan is empty and applying it will not do anything
 	IsEmpty bool
 }
+
+type Action struct {
+	// Type and name of the resource
+	ResourceType string `json:"resource_type"`
+	ResourceName string `json:"resource_name"`
+
+	Action ActionType `json:"action"`
+}
+
+func (a Action) String() string {
+	// terraform resources have the databricks_ prefix, which is not needed.
+	rtype := strings.TrimPrefix(a.ResourceType, "databricks_")
+	return strings.Join([]string{" ", string(a.Action), rtype, a.ResourceName}, " ")
+}
+
+func (c Action) IsInplaceSupported() bool {
+	return false
+}
+
+type ActionType string
+
+const (
+	ActionTypeCreate   ActionType = "create"
+	ActionTypeDelete   ActionType = "delete"
+	ActionTypeUpdate   ActionType = "update"
+	ActionTypeNoOp     ActionType = "no-op"
+	ActionTypeRead     ActionType = "read"
+	ActionTypeRecreate ActionType = "recreate"
+)
```

Comment:
Please link to the source of truth for these action names. I suspect it's somewhere in the TF source code.

Reply:
Isn't this part of the `tfjson` package btw?

Reply:
The recreate action is not a single enum in terraform. Instead it's one of the two following array of actions []{"create", "delete"} or {"delete", "create"}. To keep the interface simple on our side I opted to redefine the enum. tf-json defines public functions which have a 1:1 mapping with the enums here.  For example: `recreate` maps to `.Replace()`

source:
https://github.com/hashicorp/terraform-json/blob/0104004301ca8e7046d089cdc2e2db2179d225be/action.go#L14

Reply:
Added a comment.